### PR TITLE
Fix regtest when using new libcap-2.42

### PR DESCRIPTION
**Pull request by mgerstner**

### CODE REVIEW

File: tests/regtest.py [hunk 1]
```diff
@@ -1042,32 +1043,51 @@ def assertHasCaps(self, path, caps):
...
+		# /usr/bin/ping = cap_net_raw+ep
+		#
+		# starting from libcap-2.42 it looks like this:
+		#
+		# /usr/bin/cdrecord cap_ipc_lock,cap_sys_rawio,cap_sys_nice,cap_sys_resource=ep
+		#
+		# see bsc#1175076 comment 2.
+		# So let's be agnostic to the output format.
+
 		# output is something like "/path/to/file = cap_stuff+letters"
```
Comment (maltek) [inline tests/regtest.py:1072]: This comment is outdated by the preceding lines, no?

File: tests/regtest.py [hunk 2]
```diff
@@ -1042,32 +1043,51 @@ def assertHasCaps(self, path, caps):
...
 		for line in getcap_out.decode('utf8').splitlines():
-			# be prudent about possible spaces or equals in paths,
-			# even though it should never occur in our test
-			# environment
-			parts = line.split('=')
-			if len(parts) < 2:
+			parts = line.split()
+			if parts[0] != path:
```
Comment (maltek) [inline tests/regtest.py:1075]: Maybe check `line.startswith(path)` and then slice that away before parsing the rest of the output? Then we can get rid of the code above that validates the niceness of the path.

File: tests/regtest.py [hunk 3]
```diff
@@ -1042,32 +1043,51 @@ def assertHasCaps(self, path, caps):
...
-			cap_path = '='.join(parts[:-1]).strip()
-			if cap_path != path:
-				# not for our file
+			if len(parts) == 3 and parts[1] == '=':
+				# the old output format
+				# getcap uses a '+' to indicate capability
+				# types, while permissions uses '=', so adjust
+				# accordingly
+				expected_caps = expected_caps.replace('=', '+')
+				parts.pop(1)
```
Comment (maltek) [inline tests/regtest.py:1086]: Does the `pop` do anything here?

Comment (maltek) [reply tests/regtest.py:1086]: Ah, now I see - you're removing the thing at index 1. Could you instead move the assignment of `parts[1]` into the different branches, please?

**Comment by mgerstner** [comment]
Okay I gave the whole approach a makeover, should be more concise now.

**Pull request merged**